### PR TITLE
Remove casts

### PR DESCRIPTION
**Pull request by alsamylkin**
Related to this issue:
https://github.com/uber-go/config/issues/32

We can use different library, e.g. https://github.com/spf13/cast to do casting,
because supports more types than we do now.

### CODE REVIEW

**Comment by glibsm** [comment]
Can we write some integration tests in another test package with one said library, a-la zap benchmarks?

Would not only be good to illustrate the usage (without incurring a dependency for users), but also to make sure we remain compatible and can tweak our API to suit that use-case better.

I believe right now it involves something like calling `Value.Value()` =/

File: command_line_provider_test.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func TestCommandLineProvider_Conversion(t *testing.T) {
 
 	v := c.Get("dozen")
 	require.True(t, v.HasValue())
-	assert.Equal(t, 13, v.AsInt())
+	assert.Equal(t, "13", v.String())
```
Comment (abhinav) [inline command_line_provider_test.go]: What about

    assert.Equal(t, 13, v.Value())

Comment (alsamylkin) [inline command_line_provider_test.go]: It is going to be a string, not a integer. Same as environment variables:
```golang
assert.Equal(t, "13", v.Value())
```

Comment (abhinav) [inline command_line_provider_test.go]: Ah, okay.

**Pull request merged**